qcodes/instrument_drivers/tektronix/Keithley_2600_channels.py

```diff
@@ -311,33 +493,65 @@ def _fast_sweep(self, start: float, stop: float, steps: int,
                   'printbuffer(1, {}, {}.nvbuffer1.readings)'.format(steps,
                                                                      channel)]
 
-        self.write(self._parent._scriptwrapper(program=script, debug=True))
-        # we must wait for the script to execute
-        oldtimeout = self._parent.visa_handle.timeout
-        self._parent.visa_handle.timeout = 2*1000*steps*nplc/50 + 5000
+        return self._execute_lua(script, steps)
+
+    def _execute_lua(self, _script: list, steps: int) -> np.ndarray:
+        """
+        This is the function that sends the Lua script to be executed and
+        returns the corresponding data from the buffer.
+
+        Args:
+            _script: The Lua script to be executed.
+            steps: Number of points.
+        """
+        nplc = self.nplc()
+        linefreq = self.linefreq()
+        _time_trace_extra_visa_timeout = self._extra_visa_timeout
+        estimated_measurement_duration = 2*1000*steps*nplc/linefreq
```

Comment:
analogous to `_extra_visa_timeout`, shall the `2` become an attribute? for example, `_measurement_duration_factor`.